atom/browser/net/atom_url_loader_factory.cc

```diff
@@ -7,15 +7,24 @@
 #include <string>
 #include <utility>
 
+#include "atom/common/native_mate_converters/file_path_converter.h"
+#include "atom/common/native_mate_converters/net_converter.h"
 #include "content/public/browser/browser_thread.h"
+#include "content/public/browser/file_url_loader.h"
+#include "gin/dictionary.h"
+#include "net/base/filename_util.h"
 #include "services/network/public/cpp/url_loader_completion_status.h"
 #include "services/network/public/mojom/url_loader.mojom.h"
 
+#include "atom/common/node_includes.h"
+
 using content::BrowserThread;
 
 namespace atom {
 
-AtomURLLoaderFactory::AtomURLLoaderFactory() {}
+AtomURLLoaderFactory::AtomURLLoaderFactory(ProtocolType type,
```

Comment:
since the loader factory is unique to every handler. Can we instead have variations of the factory like `FileURLLoaderFactory`, `BufferURLLoaderFactory`, `StringURLLoaderFactory` ?

Reply:
I'm thinking of providing a universal `registerProtocol` API that can return arbitrary response in one handler, but I'm not sure how the design of the class should look like. I'll refactor the class after implementing the `Http` and `Stream` handlers.